### DISCUSSION TITLE
Merge Request comment/activity/discussion

### DISCUSSION THREAD

**Discussion by dmoshayof** [python-gitlab/python-gitlab#2631] [Q&A]
Hi,
I want to comment a MR from python script
but I didn't find any mentions for that, do you have an idea?

**Comment by unbearables** [comment]
You can comment on MR with following code:

```python
gl = # init and auth main Gitlab object
project = gl.projects.get(project_id)
mr = project.mergerequests.get(mr_iid)
mr.discussions.create({"body": "New comment"})
```
where `project_id` needs to be inputted `mr_idd`

see more at https://python-gitlab.readthedocs.io/en/stable/gl_objects/discussions.html#examples

**Comment by magicalbob** [comment]
I'd like to use API to set MR to auto-approve, something like `gitlab --private-token $GITLAB_TOKEN project-merge-request update --project-id XXX --uuid YYY --state-event ZZZ`. Is this possible? What would ZZZ be?

**Reply by nejch** [reply]
@magicalbob what exactly do you mean by "set MR to auto-approve"? Do you just mean you want to approve the MR via the API?

**Reply by magicalbob** [reply]
@nejch I mean I want to update the MR so that its as if a user has pressed the `Set to auto-merge` button, if you see what I mean? So that when a user does approve the MR it will auto merge.

**Reply by magicalbob** [reply]
So `gitlab --private-token $GITLAB_TOKEN project-merge-request merge --project-id 24448408 --iid 169 --merge-when-pipeline-succeeds True` did nearly what I wanted to do - it merged the MR, once I'd made sure all approvals were in (so MR was was ready to merge) ......

**Reply by nejch** [reply]
@magicalbob sorry for the late response, but that should be it yes. It's just that the API seems to only use the old terminology (it used to be MWPS only, but now it's expaded to auto-merge as it applies to other checks not just pipeline success).
